### ISSUE TITLE
Server does not support basic NTLM signature key exchange

### ISSUE THREAD

**Issue by ThijsBroersen** [AgNO3/jcifs-ng#135]
I am unable to connect to a share on a server. On my Windows 10 desktop I can successfully connect, but on some server (NetApp, unknown version) it does not work and I get the following exception:

jcifs.smb.SmbUnsupportedOperationException: Server does not support basic NTLM signature key exchange

Tried all kinds of negotiation-configs for the Context but without any success. What could be cause for this error? What must the server support?

I connect using 
```` 
    auth = new NtlmPasswordAuthenticator(domain, user, password)  
    val ctx  = new BaseContext(new PropertyConfiguration(p)).withCredentials(auth)
````

**Comment by ThijsBroersen** [comment]
When forcing the config to SMB1 it can connect but .getOwnerUser returns null values :S

**Comment by mbechler** [comment]
The server does not appear to support session key exchange and therefor integrity protection of the NTLM and SPNEGO authentication exchanges. 

You should be able to set _jcifs.smb.client.enforceSpnegoIntegrity=false_ to allow the connection. This will however allow some downgrade attacks, possibly allowing an attacker to make a connection use a less-secure authentication exchange than possible.

**Comment by mbechler** [comment]
Can you find out more about the server? If this is a somewhat recent one that certainly should not happen.

**Comment by mbechler** [comment]
I think I fixed this in latest master, please try whether this works for you now.

**Comment by ThijsBroersen** [comment]
Nice to hear! I will try to test it next week.

**Comment by svella** [comment]
Not sure if the "fix" for this ever made it into a release but I have a customer running into this trying to connect to their NetApp appliance running OnTap 8.2.5.

**Comment by mbechler** [comment]
The patch is in 2.1.2 and that exact error should definitely be gone. Please double check that a recent version is used, if so please open a new issue with the exact error and preferrably log file/packet trace so that we can see what flags are negotiated.

**Comment by svella** [comment]
False alarm. I had been told incorrectly that this customer had updated to a version of our product that included jcifs-ng 2.1.3 when they were actually still using a version with 2.1.1.
